src/ffi.rs

```diff
@@ -1,34 +1,34 @@
 #![allow(clippy::transmute_ptr_to_ptr)]
 #![allow(clippy::too_many_arguments)]
 
+use crate::os::{HRESULT, LPCWSTR, LPWSTR};
+pub(crate) use crate::unknown::IDxcUnknownShim;
 use com_rs::{com_interface, iid, IUnknown, IID};
 use std::ffi::c_void;
-use winapi::shared::ntdef::{LPCWSTR, LPWSTR};
-use winapi::shared::winerror::HRESULT;
 
 pub type DxcCreateInstanceProc =
     extern "system" fn(rclsid: &IID, riid: &IID, ppv: *mut *mut c_void) -> HRESULT;
 
 pub type DxcCreateInstanceProc2 = extern "system" fn(
-    malloc: *const c_void,
+    malloc: /* IMalloc */ *const c_void,
```

Comment:
Perhaps we should make these `pub type IMalloc = c_void;` (and `pub type IStream = c_void;` below) instead?

Reply:
They likely won't be used any time soon, but it's good to have all non-void types (in the actual DXC API) properly represented even if the backing object isn't implemented.